plugins/auth-backend/src/providers/types.ts

```diff
@@ -332,3 +341,10 @@ export type SAMLProviderConfig = {
 export type SAMLEnvironmentProviderConfig = {
   [key: string]: SAMLProviderConfig;
 };
+
+export type OAuthState = {
+  /* A type for the serialized value in the `state` parameter of the OAuth authorization flow
+   */
+  nonce?: string;
```

Comment:
👍  for adding this type. I think these should both be required right, i.e. no `?`?

Reply:
IIRC, I did this because the type-checker wasn't allowing me to set `nonce` and `env` to null if the de-serialization fails. I then check if these fields are empty when using the `OAuthState`. While `nonce` is mandatory, I think `env` can be optional should other users of backstage have a different way of identifying the `env` (or completely ignoring it). 
How would you recommend having a filed that could be empty/null ? is a `string  | null` type better  instead of making a field optional ?

Reply:
With them required you can reject the request early if they're not set, and then the rest of the code can be sure that they're there. I see the case for `env` being optional, but since we don't have that atm, perhaps we could keep it required for now at least?

Regarding making things optional this is the way to go usually. Don't use `null` :grin:.

Reply:
Also , there seems to be an issue with Error handling. If `nonce` or `env` were null and an Exception were thrown during parsing, the rest of the code was continued to be called instead of aborting the request processing flow.

Reply:
So the error handling might look a bit weird in the frame handler endpoint. For the error to reach the user we actually still need to return a 200 response with a script that posts to the parent frame, but the posted payload is the error instead of an auth response.

Reply:
Understood. I removed the optional fields so that the values are no longer null. Please let review the changes again :)